fortanix-sgx-tools/src/bin/ftxsgx-runner-cargo.rs

```diff
@@ -0,0 +1,111 @@
+#[macro_use]
+extern crate serde_derive;
+
+use std::env;
+use std::ffi::OsString;
+use std::fs::File;
+use std::io::Read;
+use std::process::Command;
+
+const HEAP_SIZE: u32 = 0x20000;
+const SSAFRAMESIZE: u32 = 1;
+const STACK_SIZE: u32 = 0x20000;
+const THREADS: u32 = 1;
+const DEBUG: bool = true;
+
+#[derive(Deserialize, Debug)]
+struct Metadata {
+    heap_size: Option<u32>,
+    ssaframesize: Option<u32>,
+    stack_size: Option<u32>,
+    threads: Option<u32>,
+    debug: Option<bool>,
+}
+
+#[derive(Deserialize, Debug)]
+struct Package {
+    metadata: Metadata
+}
+
+#[derive(Deserialize, Debug)]
+struct Config {
+    package: Package
+}
+
+fn main() {
+    let key = "CARGO_MANIFEST_DIR";
+    match env::var_os(key) {
+        Some(mut filepath) => {
+            filepath.push(OsString::from("/Cargo.toml"));
+            let mut file = File::open(filepath).expect("Unable to open the manifest");
+            let mut content = String::new();
+            file.read_to_string(&mut content).expect("Unable to read the manifest");
+            let config: Config = toml::from_str(&content).expect("Unable to parse the manifest");
+            let heap_size = config.package.metadata.heap_size.unwrap_or(HEAP_SIZE).to_string();
+            let ssaframesize = config.package.metadata.ssaframesize.unwrap_or(SSAFRAMESIZE).to_string();
+            let stack_size = config.package.metadata.stack_size.unwrap_or(STACK_SIZE).to_string();
+            let threads = config.package.metadata.threads.unwrap_or(THREADS).to_string();
+            let debug = config.package.metadata.debug.unwrap_or(DEBUG);
+
+            let args: Vec<String> = env::args().collect();
+
+            let mut ftxsgx_elf2sgxs_command = Command::new("ftxsgx-elf2sgxs");
```

Comment:
We can have a help option, which could mention something on the line of "ftxsgx-elf2sgxs, ftxsgx-runner etc." should be available in $PATH.

Reply:
I don't think that's necessary. This tool will always be installed alongside those.

Reply:
Not sure how command::new works, would that require Command::new("**./**ftxsgx-elf2sgxs")

Reply:
No it will look up the commands in your path